hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/util/LogCaptor.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.fixtures.util;
+
+import static java.util.Objects.requireNonNull;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.CharArrayWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.Logger;
+import org.apache.logging.log4j.core.appender.WriterAppender;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+
+/**
+ * Helper class to register a started appender on a given {@code Logger}, and parse the events at
+ * each log level as a {@code List<String>}.
+ * <p></p>
+ * This class was ported over from mono-service. When mono-service module goes away, it will be deleted from there and this
+ * will be the only copy.
+ */
+public class LogCaptor {
```

Comment:
Do we really want to use logging as a base for tests?

Reply:
In the case of freeze service, I think that it is correct to test log events specifically. Freeze log events are not just a side effect of other actions but are important actions in their own right. Tanner tells me that freeze log events are "currently important signals for us as no other metrics currently exist to provide these for us".